source-azure-blob-storage/main.go

```diff
@@ -0,0 +1,342 @@
+package main
+
+import (
+	"context"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/bloberror"
+	"github.com/estuary/connectors/filesource"
+	"github.com/estuary/flow/go/parser"
+	pf "github.com/estuary/flow/go/protocols/flow"
+)
+
+type config struct {
+	Credentials        *credentials   `json:"credentials"`
+	StorageAccountName string         `json:"storageAccountName"`
+	ContainerName      string         `json:"containerName"`
+	Parser             *parser.Config `json:"parser"`
+	MatchKeys          string         `json:"matchKeys"`
+	AscendingKeys      bool           `json:"ascendingKeys"`
+}
+
+type credentials struct {
+	AzureClientID       string `json:"azureClientID"`
+	AzureClientSecret   string `json:"azureClientSecret"`
+	AzureTenantID       string `json:"azureTenantID"`
+	AzureSubscriptionID string `json:"azureSubscriptionID"`
+	ConnectionString    string `json:"connectionString"`
+}
+
+func (c config) Validate() error {
+	if c.Credentials.ConnectionString != "" {
+		return nil
+	}
+	var requiredProperties = [][]string{
+		{"AzureTenantID", c.Credentials.AzureTenantID},
+		{"AzureClientID", c.Credentials.AzureClientID},
+		{"AzureClientSecret", c.Credentials.AzureClientSecret},
+		{"AzureSubscriptionID", c.Credentials.AzureSubscriptionID},
+		{"StorageAccountName", c.StorageAccountName},
+		{"ContainerName", c.ContainerName},
```

Comment:
Do we need a storage account name and container name if a connection string is specified?

Reply:
In that case, we need just the container name, thanks for pointing out.